components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/permission/ManagementPermissionUtil.java

```diff
@@ -81,4 +96,54 @@ public static void updateRoleUIPermission(String roleName, String[] rawPermissio
 		return permissions;
 	}
 
+    /**
+     * To call relevant event listeners when there is a failure while updating permissions of role.
+     *
+     * @param errorMessage Relevant error message.
+     * @param permissions  Permissions updated for the role.
+     * @throws UserAdminException User Admin Exception.
+     */
+    public static void handleOnUpdatePermissionsOfRoleFailure(String errorMessage, String roleName, Permission[]
+            permissions, UserStoreManager userStoreManager)
+            throws UserAdminException {
+
+        try {
+            for (UserManagementErrorEventListener listener : UserMgtDSComponent
+                    .getUserManagementErrorEventListeners()) {
+                if (listener.isEnable() && !listener.onUpdatePermissionsOfRoleFailure(
+                        UserCoreErrorConstants.ErrorMessages.ERROR_CODE_ERROR_WHILE_UPDATING_PERMISSIONS_OF_ROLE
+                                .getCode(), String.format(
+                                UserCoreErrorConstants.ErrorMessages.ERROR_CODE_ERROR_WHILE_UPDATING_PERMISSIONS_OF_ROLE
+                                        .getMessage(), errorMessage), roleName, permissions, userStoreManager)) {
+                    return;
+                }
+            }
+        } catch (UserStoreException ex) {
+            throw new UserAdminException(
+                    "Exception while executing error listeners after a failure while updating permissions of role "
+                            + roleName, ex);
+        }
+    }
+
+    /**
+     * To call the relevant listeners after updating permissions of a role.
+     *
+     * @param roleName    Name of the role.
+     * @param permissions Relevant permissions that are updated.
+     * @throws UserStoreException User Store Exception that will be thrown by relevant listeners while handling.
+     */
+    public static void handlePostUpdatePermissionsOfRole(String roleName, Permission[] permissions,
+            UserStoreManager userStoreManager) throws UserStoreException {
+
+        for (UserOperationEventListener userOperationEventListener : UserMgtDSComponent
+                .getUserOperationEventListeners()) {
+            if (userOperationEventListener instanceof AbstractUserOperationEventListener) {
+                if (!((AbstractUserOperationEventListener) userOperationEventListener)
+                        .doPostUpdatePermissionsOfRole(roleName, permissions, userStoreManager)) {
+                    return;
+                }
+            }
+        }
+    }
+
```

Comment:
Remove unnecessary new line

Reply:
As I know, normally we add the new line between file end and the last method to improve readbility.